### PR TITLE
Fixes problem with suspended app and wrong pattern matching

### LINKED ISSUE #111: Update commit and title default message to contain used pattern if check fails

### PR DESCRIPTION
**Pull request by The-Huginn**
Please review @petrberan @xstefank 

Resolves #111

### CODE REVIEW

File: src/main/java/io/xstefank/wildfly/bot/LifecycleProcessor.java [hunk 1]
```diff
@@ -100,11 +100,36 @@ void onStart(@Observes StartupEvent event) {
...
         } catch (IOException e) {
-            LOG.warn("Unable to verify rules in repository. Use debug log level for more details.");
-            LOG.debug("Unable to verify rules in repository.", e);
+            unableToStartLog(Logger.Level.WARN, () -> {
+                LOG.warn("Unable to verify rules in repository. Use debug log level for more details.");
+                LOG.debug("Unable to verify rules in repository.", e);
+            });
+        } catch (IllegalStateException e) {
+            if (e.getCause() instanceof HttpException && e.getCause().getMessage().contains("suspended")) {
+                unableToStartLog(Logger.Level.WARN, () -> LOG.warnf("Your installation has been suspended. %s will not be able to operate correctly.", wildFlyBotConfig.githubName()));
```
Comment (xstefank) [inline src/main/java/io/xstefank/wildfly/bot/LifecycleProcessor.java:109]: log also some identification of the installation

File: src/main/java/io/xstefank/wildfly/bot/LifecycleProcessor.java [hunk 2]
```diff
@@ -100,11 +100,36 @@ void onStart(@Observes StartupEvent event) {
...
+    void unsuspendedInstallation(@Installation.Unsuspend GHEventPayload.Installation installationPayload) {
+        GHAppInstallation installation = installationPayload.getInstallation();
+        LOG.infof("%s has been unsuspended for following installation id: %d and will start to listen for new Pull Request changes.", wildFlyBotConfig.githubName(), installation.getAppId());
+    }
+
+    private void unableToStartLog(Logger.Level logLevel, Runnable log) {
+        LOG.logf(logLevel, "############## IMPORTANT ##################");
+        log.run();
+        LOG.logf(logLevel, "%s might not work as expected, please consider fixing the cause of the problem", wildFlyBotConfig.githubName());
+        LOG.logf(logLevel, "###########################################");
```
Comment (xstefank) [inline src/main/java/io/xstefank/wildfly/bot/LifecycleProcessor.java:130]: not sure if this is needed

**Review by xstefank — CHANGES_REQUESTED** [review]

File: src/test/java/io/xstefank/wildfly/bot/InstallationEventTest.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    private static final java.util.logging.Logger rootLogger = LogManager.getLogManager().getLogger("io.xstefank.wildfly.bot");
+    private static final InMemoryLogHandler inMemoryLogHandler = new InMemoryLogHandler(
+            record -> record.getLevel().intValue() >= Level.ALL.intValue());
+
+    static {
+        rootLogger.addHandler(inMemoryLogHandler);
+    }
+
+    @RegisterExtension
+    static QuarkusTestExtension TEST = new QuarkusTestExtension();
```
Comment (xstefank) [inline src/test/java/io/xstefank/wildfly/bot/InstallationEventTest.java:31]: why do you need this?

**Review by xstefank — CHANGES_REQUESTED** [review]

**Comment by The-Huginn** [comment]
Resolves #111

File: src/test/java/io/xstefank/wildfly/bot/InstallationEventTest.java (hunk 3 above)
Comment (The-Huginn) [reply src/test/java/io/xstefank/wildfly/bot/InstallationEventTest.java:31]: You are right, there is no need for that. Removing that

File: src/test/java/io/xstefank/wildfly/bot/PRCommitCheckTest.java [hunk 4]
```diff
@@ -49,8 +47,7 @@ void testFailedCommitCheck() throws IOException {
                 .then().github(mocks -> {
                     GHRepository repo = mocks.repository(TEST_REPO);
                     Util.verifyFormatFailure(repo, gitHubJson, "commit");
-                    Util.verifyFailedFormatComment(mocks, gitHubJson,
-                            String.format("- For commit: \"%s\" (%s) - " + DEFAULT_COMMIT_MESSAGE, INVALID_COMMIT_MESSAGE, gitHubJson.commitSHA()));
+                    Util.verifyFailedFormatComment(mocks, gitHubJson,"- " + String.format(DEFAULT_COMMIT_MESSAGE, "\\[WFLY-\\d+]\\s+.*|WFLY-\\d+\\s+.*"));
```
Comment (xstefank) [inline src/test/java/io/xstefank/wildfly/bot/PRCommitCheckTest.java:50]: we should probably try to get this from config somewhere?

**Review by xstefank — APPROVED** [review]

**Pull request merged**